packages/compass-aggregations/src/modules/collections-fields.spec.ts

```diff
@@ -0,0 +1,284 @@
+          'test.coll2': {
+            isLoading: false,
+            fields: [],
+            type: 'collection',
+          },
+        });
+      });
+
+      it('does not remove existing collection data when setting updated list', async function () {
+        // set the collection
+        await store.dispatch(
+          setCollections([
+            {
+              name: 'test.coll',
+              type: 'collection',
+            },
+          ]) as any
+        );
+        // set the collection fields
+        await store.dispatch(
+          setCollectionFields('test.coll', 'collection', ['a', 'b', 'c']) as any
+        );
+
+        await store.dispatch(
+          setCollections([
+            {
+              name: 'test.coll',
+              type: 'collection',
+            },
+            {
+              name: 'test.coll2',
+              type: 'collection',
+            },
+          ]) as any
+        );
+        expect(store.getState().collectionsFields).to.deep.equal({
+          'test.coll': {
+            isLoading: false,
+            fields: ['a', 'b', 'c'],
+            type: 'collection',
+          },
+          'test.coll2': {
+            isLoading: false,
+            fields: [],
+            type: 'collection',
+          },
+        });
+      });
+    });
+
+    context('setCollectionFields', function () {
+      it('sets collection fields', async function () {
+        await store.dispatch(
+          setCollectionFields('test.coll', 'collection', ['a', 'b', 'c']) as any
+        );
+        expect(store.getState().collectionsFields).to.deep.equal({
+          'test.coll': {
+            isLoading: false,
+            fields: ['a', 'b', 'c'],
+            type: 'collection',
+          },
+        });
+      });
+    });
+
+    context('fetchCollectionFields', function () {
+      it('does nothing if fields are already fetched', async function () {
+        await store.dispatch(
+          setCollectionFields('test.coll', 'collection', ['a', 'b', 'c']) as any
+        );
+        await store.dispatch(fetchCollectionFields('test.coll') as any);
+        expect(store.getState().collectionsFields).to.deep.equal({
+          'test.coll': {
+            isLoading: false,
+            fields: ['a', 'b', 'c'],
+            type: 'collection',
+          },
+        });
+      });
+
+      it('uses sample to fetch schema of a view', async function () {
+        await store.dispatch(
+          setCollections([
+            {
+              name: 'listings',
+              type: 'view',
+            },
+          ]) as any
+        );
+
+        const spy = sinon.spy(
+          (ns: string, stage: Document, options: Document) => {
+            expect(ns).to.equal('test.listings');
+            expect(stage).to.deep.equal({ size: 1 });
+            expect(options).to.deep.equal({ maxTimeMS: 5000 });
+            return Promise.resolve([{ _id: 12, name: 'test' }]);
```

Comment:
Question: Is this a common pattern? For this I probably would have done my expects on the arguments of the calls to the spy function?
Nothing to do here as I am just looking for new patterns to look into :D

Reply:
Good point. Umm, well I added these args in the spy as ts was complaining when I tried to access the arguments of spy `spy.getCalls()[0].args[0]`. And when I added the args, then unused args error. So I ended I doing this as I am expecting only one call to be made.

Normally I'd do this outside of the spy.